### PR TITLE
Add Laravel packages on creating a new project

### PR DESCRIPTION
**Pull request by regiszanandrea**
Hi,

This PR references to idea [#1987](https://github.com/laravel/ideas/issues/1987)

This PR add an option to laravel/installer new command that install Laravel's packages after project creation. 
Only Laravel Packages are allowed, excluding paid projects.

The option name is `with` but could be other.

An example:
`$ laravel new blog --with=ui,horizon,telescope,socialite`

This will install `ui`, `telescope` and `socialite` packages and run their `install` commands like: `php artisan telescope:install`.

The package `ui` will be installed in `require-dev` section of composer.json respecting the official documentation.

Also, I improved some code style on NewCommand Class.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please fix the phpdoc, in particular, to match Laravel's standards.

Don't worry about the rest of the code style though. It gets auto-fixed on merge. Easier for everyone that way. Contributors don't need to worry, and neither do maintainers.

File: src/NewCommand.php [hunk 1]
```diff
@@ -243,4 +284,73 @@ protected function findComposer()
 
         return 'composer';
     }
+
+    /**
+     * @param string $packages
+     * @param string $composer
+     * @return string
+     */
+    private function getLaravelPackages(string $packages, string $composer)
```
Comment (GrahamCampbell) [inline src/NewCommand.php]: Please remove the string type here

File: src/NewCommand.php [hunk 2]
```diff
@@ -243,4 +284,73 @@ protected function findComposer()
...
+
+        return $composerPackages;
+    }
+
+    /**
+     * @param string $packages
+     * @param string $composer
+     * @return string
+     */
+    private function getLaravelDevPackages(string $packages, string $composer)
```
Comment (GrahamCampbell) [inline src/NewCommand.php]: And these

File: src/NewCommand.php [hunk 3]
```diff
@@ -243,4 +284,73 @@ protected function findComposer()
...
+        }
+
+        return $composerPackages;
+    }
+
+    /**
+     * @param string $packages
+     * @return string
+     */
+    private function getLaravelInstallCommands(string $packages)
```
Comment (GrahamCampbell) [inline src/NewCommand.php]: and this one

File: src/NewCommand.php [hunk 4]
```diff
@@ -243,4 +284,73 @@ protected function findComposer()
...
+
+        if ($composerPackages !== '') {
+            return $composer . ' require --dev' . $composerPackages;
+        }
+
+        return $composerPackages;
+    }
+
+    /**
+     * @param string $packages
```
Comment (GrahamCampbell) [inline src/NewCommand.php]: Please put 2 spaces before and after the word string in param docs, in all cases.

**Comment by regiszanandrea** [comment]
@GrahamCampbell Like 492ae3c?

File: src/NewCommand.php (hunk 4 above)
Comment (regiszanandrea) [reply src/NewCommand.php]: I think it's ok now, right?

File: src/NewCommand.php [hunk 5]
```diff
@@ -81,6 +107,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
             }, $commands);
         }
 
+        if ($input->getOption('with')) {
```
Comment (nunomaduro) [inline src/NewCommand.php]: Here you can do `if ($with = $input->getOption('with')) {` and replace all `$input->getOption('with')` bellow with `$with`.

File: src/NewCommand.php [hunk 6]
```diff
@@ -243,4 +283,73 @@ protected function findComposer()
...
+     * @return  string
+     */
+    private function getLaravelInstallCommands($packages)
+    {
+        $laravelPackages = explode(',', $packages);
+
+        $packagesInstallCommands = '';
+
+        foreach ($laravelPackages as $laravelPackage) {
+            if (array_key_exists($laravelPackage, self::LARAVEL_PACKAGES_INSTALL_COMMAND)) {
```
Comment (nunomaduro) [inline src/NewCommand.php]: `array_key_exists` is deprecated: https://www.php.net/manual/es/migration74.deprecated.php#migration74.deprecated.core.array-key-exists-objects.

File: src/NewCommand.php [hunk 7]
```diff
@@ -243,4 +283,73 @@ protected function findComposer()
 
         return 'composer';
     }
+
+    /**
+     * @param  string  $packages
+     * @param  string  $composer
+     * @return  string
+     */
+    private function getLaravelPackages($packages, $composer)
```
Comment (nunomaduro) [inline src/NewCommand.php]: This should be `protected` instead.

File: src/NewCommand.php [hunk 8]
```diff
@@ -243,4 +283,73 @@ protected function findComposer()
...
+
+        return $composerPackages;
+    }
+
+    /**
+     * @param  string  $packages
+     * @param  string  $composer
+     * @return  string
+     */
+    private function getLaravelDevPackages($packages, $composer)
```
Comment (nunomaduro) [inline src/NewCommand.php]: This should be `protected` instead.

File: src/NewCommand.php [hunk 9]
```diff
@@ -243,4 +283,73 @@ protected function findComposer()
...
+        }
+
+        return $composerPackages;
+    }
+
+    /**
+     * @param  string  $packages
+     * @return  string
+     */
+    private function getLaravelInstallCommands($packages)
```
Comment (nunomaduro) [inline src/NewCommand.php]: This should be `protected` instead.

File: src/NewCommand.php (hunk 7 above)
Comment (nunomaduro) [reply src/NewCommand.php]: The only difference between `getLaravelPackages` and `getLaravelDevPackages` is the flag `--dev` and the `list of packages`.

Maybe consider making a generic function like `getPackages($composer , $with, $dev = true)`.

**Comment by regiszanandrea** [comment]
@GrahamCampbell Fixed, the checks are returning ok.

File: src/NewCommand.php (hunk 6 above)
Comment (regiszanandrea) [reply src/NewCommand.php]: Do you recommend use `isset`?

Comment (nunomaduro) [reply src/NewCommand.php]: Yes.

**Comment by regiszanandrea** [comment]
@nunomaduro I changed the code by your comments. Thanks so much <3

**Comment by nunomaduro** [comment]
@regiszanandrea No worries. I would like to review little more your pull request, but before we do it. Let's make sure @taylorotwell would like to have this on the installer.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by mattstauffer** [comment]
@regiszanandrea If you're still interested in this functionality, it would make sense for us to add it to Lambo. However, I'd recommend we wait until I finish re-writing it in PHP, which I hope to complete in the next few weeks.

**Comment by regiszanandrea** [comment]
@mattstauffer Let's do this 💯

**Pull request closed without merging**